lambdas/drain_ecs_container_instance/drain_ecs_container_instance.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+"""
+This task tries to ensure graceful termination of ECS container instances.
+
+The SNS topic "ec2_terminating" receives messages telling us about terminating EC2 instances.
+If the terminating instance is part of an ECS cluster, it drains the ECS tasks on the instance.
+"""
+import json
+import pprint
+import time
+
+import boto3
+
+from sns_utils import publish_sns_message
+
+
+def set_container_instance_to_draining(ecs_client, cluster_arn, ecs_container_instance_arn):
+    ecs_client.update_container_instances_state(
+        cluster=cluster_arn,
+        containerInstances=[
+            ecs_container_instance_arn,
+        ],
+        status='DRAINING'
+    )
+
+
+def continue_lifecycle_action(asg_client, asg_group_name, ec2_instance_id, lifecycle_hook_name):
+    response = asg_client.complete_lifecycle_action(
+        LifecycleHookName=lifecycle_hook_name,
+        AutoScalingGroupName=asg_group_name,
+        LifecycleActionResult='CONTINUE',
+        InstanceId=ec2_instance_id)
+    pprint.pprint(response)
+
+
+def get_ecs_info_from_tags(ec2_client, ec2_instance_id):
+    ec2_instance_info = ec2_client.describe_instances(InstanceIds=[
+        ec2_instance_id,
+    ])
+    tags = ec2_instance_info['Reservations'][0]['Instances'][0]['Tags']
+    ecs_container_instance_arns = [x['Value'] for x in tags if x['Key'] == 'containerInstanceArn']
+    cluster_arns = [x['Value'] for x in tags if x['Key'] == 'clusterArn']
```

Comment:
An alternative way we could do this would be to unpack this back into a proper dictionary object:

```python
tag_dict = {t['Key']: t['Value'] for t in tags}
```

You could then query individual items like a dict:

```python
ecs_container_instance_arn = tag_dict['containerInstanceArn']
cluster_arn = tag_dict['clusterArn']
```

and catch the `KeyError` if the key is missing. Might that be simpler?